core/js/src/main/scala/cats/effect/unsafe/FiberMonitor.scala

```diff
@@ -50,20 +56,22 @@ private final class ES2021SuspendedFiberBag extends SuspendedFiberBag {
  * A no-op implementation of an unordered bag used for tracking asynchronously suspended fiber
  * instances on Scala.js. This is used as a fallback.
  */
-private final class NoOpSuspendedFiberBag extends SuspendedFiberBag {
-  override def monitor(key: AnyRef, fiber: IOFiber[_]): Unit = ()
+private final class NoOpFiberMonitor extends FiberMonitor {
+  override def monitorSuspended(key: AnyRef, fiber: IOFiber[_]): Unit = ()
 }
 
-private[effect] object SuspendedFiberBag {
+private[effect] object FiberMonitor {
 
-  // Only exists for source compatibility with JVM code.
-  @nowarn("cat=unused-params")
-  def apply(compute: WorkStealingThreadPool): SuspendedFiberBag =
-    apply()
-
-  def apply(): SuspendedFiberBag =
-    if (LinkingInfo.developmentMode && IterableWeakMap.isAvailable)
-      new ES2021SuspendedFiberBag
-    else
-      new NoOpSuspendedFiberBag
+  def apply(compute: ExecutionContext): FiberMonitor = {
+    if (LinkingInfo.developmentMode && IterableWeakMap.isAvailable) {
+      if (compute.isInstanceOf[FiberAwareExecutionContext]) {
+        val faec = compute.asInstanceOf[FiberAwareExecutionContext]
+        new ES2021FiberMonitor(faec)
+      } else {
+        new NoOpFiberMonitor()
```

Comment:
Going on your comment above, can this be `null`? I don't have any strong feeling about this either way. Although it's worth pointing out, tracking suspended fibers is valuable even without being able to track live fibers, but probably not the other way around. I think the question is, if someone else plugs their custom EC into the `IORuntime`, do we make it fiber-aware or not? Not worth dwelling on too much since the EC options are pretty dry anyway 😆 

```suggestion
        new ES2021FiberMonitor(null)
```